tg_bot/modules/locks.py

```diff
@@ -94,6 +98,77 @@ def locktypes(bot: Bot, update: Update):
     update.effective_message.reply_text("\n - ".join(["Locks: "] + list(LOCK_TYPES) + list(RESTRICTION_TYPES)))
 
 
+@run_async
+@user_admin
+@loggable
+def add_whitelist(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+    user = update.effective_user  # type: Optional[User]
+    message = update.effective_message  # type: Optional[Message]
+    entities = message.parse_entities([MessageEntity.URL])
+    added = []
+    for url in entities.values():
+        if sql.add_whitelist(chat.id, url):
+            added.append(url)
+    if added:
+        message.reply_text("Added to whitelist:\n- "+'\n- '.join(added))
+        return "<b>{}:</b>" \
+               "\n#WHITELIST" \
+               "\n<b>Admin:</b> {}" \
+               "\nWhitelisted:\n<pre>- {}</pre>".format(html.escape(chat.title),
+                                                         mention_html(user.id, user.first_name),
+                                                         html.escape('\n- '.join(added)))
+    else:
+        message.reply_text("No URLs were added to the whitelist")
+        return ""
+
+
+@run_async
+@user_admin
+@loggable
+def remove_whitelist(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+    user = update.effective_user  # type: Optional[User]
+    message = update.effective_message  # type: Optional[Message]
+    entities = message.parse_entities(MessageEntity.URL)
+    removed = []
+    for url in entities.values():
+        if sql.remove_whitelist(chat.id, url):
+            removed.append(url)
+    if removed:
+        message.reply_text("Removed from whitelist:\n<pre>- {}</pre>".format(html.escape('\n- '.join(removed))),
+                           parse_mode=ParseMode.HTML)
+        return "<b>{}:</b>" \
+               "\n#UNWHITELIST" \
+               "\n<b>Admin:</b> {}" \
+               "\nRemoved from whitelist:\n<pre>- {}</pre>".format(html.escape(chat.title),
+                                                                    mention_html(user.id, user.first_name),
+                                                                    html.escape('\n- '.join(removed)))
+    else:
+        message.reply_text("Could not remove URL from whitelist or URL not found.")
+        return ""
+
+
+@run_async
+def list_whitelist(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+    message = update.effective_message  # type: Optional[Message]
+    all_whitelisted = sql.get_whitelist(chat.id)
+
+    if not all_whitelisted:
+        message.reply_text("No URLs are whitelisted here!")
+        return
+
+    whitelist_string = "Whitelisted URLs:\n- "+'\n- '.join(all_whitelisted)
```

Comment:
Why arent you using `<code>`? would make it easier to remove the urls

Reply:
`<code>` can be copied with a single tap, but Telegram doesn't give you any feedback of what was copied. If you have a list of links, you won't know if you tapped the correct one (fat finger). By using links, copying is almost as easy by tap+holding, but you also get the feedback of which link was tapped.

When removing from the whitelist, I used <pre> to prevent others from clicking (will change to individual `<code>` blocks), but when whitelisting, I believe leaving the links clickable is better.